### PR TITLE
feat: replace deprecated chip component with tag

### LINKED ISSUE #20487: Replace deprecated Chip component with Tag from the component-library
### Description

Currently, the extension is using an outdated `Chip` component, which needs to be replaced with the new `Tag` component.

This is a massive undertaking by itself and creating a single PR would be too large. Smaller PRs can be submitted against this issue to ensure easier review and gradual improvements.

### Technical Details

- Replace instances of `Chip` component (`ui/components/ui/chip/chip.js`) with `Tag` component (`ui/components/component-library/tag/tag.tsx`)
- Component APIs are slightly different so ensure all props have been migrated appropriately

### Acceptance Criteria

- Instances of `Chip` component are completely replaced with the new `Tag` component
- The component APIs are updated to reflect the changes in the new `Chip`` component and there is no functional changes or visual regression
- Each Pull Request (PR) should include **no more than 3 files**
- The code changes should pass Jest tests, linting, and Storybook without any errors.
- The PR must include before and after screenshots of the UI to ensure there are no visual regressions.

If the acceptance criteria is not met, PRs may be closed.

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with running the extension locally, have knowledge of React, component props, Jest tests, linting, and Storybook, and want to contribute to improving the cohesiveness of UI in the extension

### PR DESCRIPTION
**Pull request by nickpismenkov**
## **Description**

Replaced deprecated chip component with tag in recovery-phrase-chips component

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24594?quickstart=1)

## **Related issues**

Fixes: #20487

## **Manual testing steps**

1. Install extension
2. Click "Create a new wallet"
3. Enter password
4. Click "Secure my wallet"
5. Click "Reveal seed phrase"

### **Before**
Screenshot 2024-05-21 at 19 29 07

### **After**
Screenshot 2024-05-21 at 19 22 23

https://github.com/MetaMask/metamask-extension/assets/8112138/2a97ccbe-2a27-4488-b0a5-86159020f0bd

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by georgewrmarshall** [comment]
Hey @nickpismenkov, could you please make sure the PR description is complete and includes the manual testing steps for the reviewers? Thanks!

**Comment by nickpismenkov** [comment]
@georgewrmarshall Hi. I completed description. Could you please help with e2e failed tests?

File: ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
 import React from 'react';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
-import Chip from '../../../components/ui/chip';
-import Box from '../../../components/ui/box';
-import Typography from '../../../components/ui/typography';
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.js:6]: Deprecated components

File: ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 import { ChipWithInput } from '../../../components/ui/chip/chip-with-input';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 import {
-  TypographyVariant,
+  TextVariant,
   BorderStyle,
-  Size,
-  DISPLAY,
+  BorderRadius,
+  Display,
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.js:17]: Replacing deprecated helpers

File: ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.stories.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+const Template = (args) => <RecoveryPhraseChips {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
+  phraseRevealed: false,
+};
+
+export const PhraseRevealed = Template.bind({});
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.stories.js:56]: Adding a Storybook story for easier isolated development

https://github.com/MetaMask/metamask-extension/assets/8112138/6aad55ed-02bf-4ab9-b829-804429fec421

File: ui/pages/onboarding-flow/recovery-phrase/index.scss [hunk 4]
```diff
@@ -21,58 +21,6 @@
...
-    display: flex;
-    flex-direction: row;
-    align-items: center;
-    text-align: center;
-
-    &__number {
-      font-size: design-system.$font-size-h5;
-    }
-  }
-
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/recovery-phrase/index.scss:75]: Removing unused styles and moving some to https://github.com/MetaMask/metamask-extension/pull/24594/files#diff-58ef2d79c613500bbe5a15545874fbeb5a4d07a22d3796a9e9a1d9463a592347

File: ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.test.js [hunk 5]
```diff
@@ -19,7 +19,7 @@ describe('Recovery Phrase Chips Component', () => {
 
     const recoveryPhraseChips = queryByTestId('recovery-phrase-chips');
     expect(recoveryPhraseChips).not.toHaveClass(
-      'recovery-phrase__chips--hidden',
+      'recovery-phrase-chips__chips--hidden',
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.test.js:22]: Updating classnames to reflect whats in javascript

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @nickpismenkov,

I did some more digging, and I think this task is a bit more complex than I initially thought. I've refactored it to align with our design system standards. 

I've created a couple of other PRs that will need to be merged before this one:
- https://github.com/MetaMask/metamask-extension/pull/24781
- https://github.com/MetaMask/metamask-extension/pull/24780

Once the above PRs are merged, we can rebase and re-test. Thanks for your contribution and patience.

Blocking until above PRs have been merged

**Pull request closed without merging**